### DISCUSSION TITLE
Can I bind resolver to interface type, so it became default resolver for Interface inheritors?

### DISCUSSION THREAD

**Discussion by Sergei-Rudenkov** [mirumee/ariadne#725] [Q&A]
I have schema:

```
interface MyInterface {
    name: String
}
```

I would like to define resolver to all Objects that use this interface:

```
my_interface = InterfaceType("MyInterface")
@my_interface.field("name")
def resolve():
    #...
```

How ever this resolver doesn't bind to Objects that implements `MyInterface`. Is support of this functionality planned in Ariadne?

**Comment by rafalp** [comment]
Are you passing `my_interface` to `make_executable_schema`? The behavior you are describing is supposed to work in Ariadne.

**Reply by Sergei-Rudenkov** [reply]
Thanks for your answer! I investigated a bit more
It happens that resolver doesn't invocate because I have directive on the same field aand it looks like:
```
class MetadataDirective(SchemaDirectiveVisitor):
    def visit_field_definition(
        self,
        field: GraphQLField,
        object_type: GraphQLObjectType | GraphQLInterfaceType,
    ) -> GraphQLField:
        original_resolver = field.resolve or default_field_resolver
```

however field doesn't have resolver, resolver has the interface. 
I found resolver in field attributes:
```
        original_resolver = field.resolve or default_field_resolver
        if object_type.interfaces and object_type.interfaces[0].fields.get('metadata'):
            original_resolver = object_type.interfaces[0].fields['metadata'].resolve or default_field_resolver
```

Do you have better solution for me?

PS, I tried to put directive on field inside Interface. But it doesn't work any way.

**Reply by rafalp** [reply]
Are you passing my_interface to make_executable_schema? Directives are applied after bindables, so field.resolve should be set by InterfaceType and then wrapped by directive.
